nixos/modules/misc/version.nix

```
@@ -121,22 +121,29 @@ in
       default = cfg.release;
       defaultText = literalExpression "config.${opt.release}";
       description = lib.mdDoc ''
-        Every once in a while, a new NixOS release may change
-        configuration defaults in a way incompatible with stateful
-        data. For instance, if the default version of PostgreSQL
-        changes, the new version will probably be unable to read your
-        existing databases. To prevent such breakage, you should set the
-        value of this option to the NixOS release with which you want
-        to be compatible. The effect is that NixOS will use
-        defaults corresponding to the specified release (such as using
-        an older version of PostgreSQL).
-        It’s perfectly fine and recommended to leave this value at the
-        release version of the first install of this system.
-        Changing this option will not upgrade your system. In fact it
-        is meant to stay constant exactly when you upgrade your system.
-        You should only bump this option, if you are sure that you can
-        or have migrated all state on your system which is affected
-        by this option.
+        This option defines the first version of NixOS you have installed on this particular machine,
+        and is used to maintain compatibility with data not managed by NixOS configuration (e.g. databases)
+        that was created by applications running on older NixOS versions.
+
+        For example, if NixOS version XX.YY ships with AwesomeDB version N by default, and is then
+        upgraded to version XX.YY+1, which ships AwesomeDB version N+1, the existing databases
+        may no longer be compatible, causing applications to fail, or even data loss.
```

```suggestion
        may no longer be compatible, causing application failure or even data loss.
```

---

I think I prefer active voice here. Might be just me, but it sounds more urgent.

---

```suggestion
        may no longer be compatible, causing applications to fail, or even lead to data loss.
```

Without an active verb the second part does not mesh with the first.

---

"leading to" then probably?

---

Yes, "leading to" flows correctly here. Added.